src/main/java/com/autotune/common/kafka/KruizeKafkaProducer.java

```diff
@@ -0,0 +1,156 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.autotune.common.kafka;
+
+import com.autotune.operator.KruizeDeploymentInfo;
+import com.autotune.utils.KruizeConstants;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * KruizeKafkaProducer class is responsible for producing messages to Kafka topics.
+ * It provides methods to send different types of messages, such as recommendations, errors, and summaries.
+ */
+public class KruizeKafkaProducer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KruizeKafkaProducer.class);
+
+    // Singleton Kafka Producer Instance
+    private static final KafkaProducer<String, String> producer = new KafkaProducer<>(getProducerProperties());
+
+    /**
+     * Retrieves Kafka producer properties.     *
+     * @return Properties object containing Kafka producer configuration.
+     */
+    private static Properties getProducerProperties() {
+        Properties producerProps = new Properties();
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, KruizeDeploymentInfo.kafka_bootstrap_servers);
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        producerProps.put(ProducerConfig.ACKS_CONFIG, "all");
+        return producerProps;
+    }
+
+    /**
+     * Sends a message to the specified Kafka topic.
+     *
+     * @param topic   The Kafka topic to send the message to.
+     * @param payload The message payload.
+     */
+    private static void sendMessage(String topic, String payload) {
+        try {
+            RecordMetadata metadata = producer.send(new ProducerRecord<>(topic, payload))
+                    .get(5, TimeUnit.SECONDS); //todo : set the timeout value via ENV
+            // todo: get the status of message whether its delivered or failed
+            LOGGER.debug("Message sent successfully to topic {} at partition {} and offset {}",
+                    metadata.topic(), metadata.partition(), metadata.offset());
+        } catch (TimeoutException te) {
+            LOGGER.error("Kafka timeout while sending message to topic {}: {}", topic, te.getMessage());
+        } catch (Exception e) {
+            LOGGER.error("Error sending message to Kafka topic {}: {}", topic, e.getMessage(), e);
```

Comment:
Please use constants for all the strings

Reply:
Updated now